### PR TITLE
Mpileup intervals

### PR DESCRIPTION
**Pull request by nickhsmith**
Add Intervals to bcftools mpileup using the `-T` option. Add approrpriate tests using sarscov2 bed file

### CODE REVIEW

**Comment by nickhsmith** [comment]
Does anyone have a suggestion for how to control for the htslib dependency of `bcftools`?

The Docker/Singularity version gives this:
`# This file was produced by bcftools stats (1.15.1+htslib-1.15) and can be plotted using plot-vcfstats.`

The Conda version gives this:
`# This file was produced by bcftools stats (1.15.1+htslib-1.16) and can be plotted using plot-vcfstats.`

**Review by maxulysse — APPROVED** [review]

File: tests/modules/nf-core/bcftools/mpileup/test.yml [hunk 1]
```diff
@@ -18,23 +18,23 @@
...
 - name: bcftools mpileup test_bcftools_mpileup_intervals
   command: nextflow run ./tests/modules/nf-core/bcftools/mpileup -entry test_bcftools_mpileup_intervals -c ./tests/config/nextflow.config     -c ./tests/modules/nf-core/bcftools/mpileup/nextflow.config
-  tags:
-    - bcftools/mpileup
-    - bcftools
-  files:
+  tags:                                                                                                                 
+    - bcftools/mpileup                                                                                                  
+    - bcftools                                                                                                          
+  files:
```
Comment (maxulysse) [inline tests/modules/nf-core/bcftools/mpileup/test.yml:34]:
```suggestion
  tags:
    - bcftools/mpileup
    - bcftools
  files:
```

File: modules/nf-core/bcftools/mpileup/main.nf [hunk 2]
```diff
@@ -2,10 +2,10 @@ process BCFTOOLS_MPILEUP {
     tag "$meta.id"
     label 'process_medium'
 
-    conda (params.enable_conda ? "bioconda::bcftools=1.15.1" : null)
+    conda (params.enable_conda ? "bioconda::bcftools=1.16" : null)
     container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
-        'https://depot.galaxyproject.org/singularity/bcftools:1.15.1--h0ea216a_0':
-        'quay.io/biocontainers/bcftools:1.15.1--h0ea216a_0' }"
+		'https://depot.galaxyproject.org/singularity/bcftools:1.16.1--hfe4b78e_1':
```
Comment (maxulysse) [inline modules/nf-core/bcftools/mpileup/main.nf:7]:
```suggestion
        'https://depot.galaxyproject.org/singularity/bcftools:1.16.1--hfe4b78e_1':
```

File: modules/nf-core/bcftools/mpileup/main.nf [hunk 3]
```diff
@@ -2,13 +2,13 @@ process BCFTOOLS_MPILEUP {
     tag "$meta.id"
     label 'process_medium'
 
-    conda (params.enable_conda ? "bioconda::bcftools=1.15.1" : null)
+    conda (params.enable_conda ? "bioconda::bcftools=1.16" : null)
     container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
-        'https://depot.galaxyproject.org/singularity/bcftools:1.15.1--h0ea216a_0':
-        'quay.io/biocontainers/bcftools:1.15.1--h0ea216a_0' }"
+        'https://depot.galaxyproject.org/singularity/bcftools:1.16.1--hfe4b78e_1':
```
Comment (maxulysse) [inline modules/nf-core/bcftools/mpileup/main.nf:7]:
```suggestion
        'https://depot.galaxyproject.org/singularity/bcftools:1.16--hfe4b78e_1':
```

**Comment by maxulysse** [comment]
@nickhsmith sorry for stepping in, but tiny typo on the singularity container definition, figured it would be faster just to fix it.

**Pull request merged**